ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirebaseMessagingService.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.persistence.remote.firebase
+
+import com.google.android.ground.persistence.sync.SurveySyncService
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import timber.log.Timber
+
+@AndroidEntryPoint
+class FirebaseMessagingService : FirebaseMessagingService() {
+
+  @Inject lateinit var surveySyncService: SurveySyncService
+
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
```

Comment:
@shobhitagarwal1612 Discussed moving this logic to a delegate to facilitate testing.